Made ADReynoldsNumberMaterial take AD hydraulic diameter

joshuahansel opened the pull request:
Closes #20253

moosebuild commented:
Job [Coverage](https://civet.inl.gov/job/951678/) on 49059be wanted to post the following:


## Framework coverage

Coverage did not change


## Modules coverage

### Thermal hydraulics

<table>
  <thead>
    <tr>
      <th></th>
      <th>db314b</th>
      <th colspan=4>#20254 49059b</th>
    </tr>
    <tr>
      <th></th>
      <th>Total</th>
      <th>Total</th>
      <th>+/-</th>
      <th>New</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td><b>Rate</b></td>
      <td>90.34%</td>
      <td>90.34%</td>
      <td>-</td>
      <td>100.00%</td>
    </tr>
    <tr>
      <td><b>Hits</b></td>
      <td>14918</td>
      <td>14918</td>
      <td>-</td>
      <td>1</td>
    </tr>
    <tr>
      <td><b>Misses</b></td>
      <td>1596</td>
      <td>1596</td>
      <td>-</td>
      <td>0</td>
    </tr>
  </tbody>
</table>

[Diff coverage report](https://mooseframework.inl.gov/docs/PRs/20254/coverage_diff/thermal_hydraulics.html)

[Full coverage report](https://mooseframework.inl.gov/docs/PRs/20254/coverage/thermal_hydraulics)

## Full coverage reports

<details>
<summary>Reports</summary>

- [`framework`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/framework)
- [`chemical_reactions`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/chemical_reactions)
- [`combined`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/combined)
- [`contact`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/contact)
- [`external_petsc_solver`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/external_petsc_solver)
- [`fluid_properties`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/fluid_properties)
- [`fsi`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/fsi)
- [`functional_expansion_tools`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/functional_expansion_tools)
- [`geochemistry`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/geochemistry)
- [`heat_conduction`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/heat_conduction)
- [`level_set`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/level_set)
- [`misc`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/misc)
- [`navier_stokes`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/navier_stokes)
- [`peridynamics`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/peridynamics)
- [`phase_field`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/phase_field)
- [`porous_flow`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/porous_flow)
- [`ray_tracing`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/ray_tracing)
- [`rdg`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/rdg)
- [`reactor`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/reactor)
- [`richards`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/richards)
- [`stochastic_tools`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/stochastic_tools)
- [`tensor_mechanics`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/tensor_mechanics)
- [`thermal_hydraulics`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/thermal_hydraulics)
- [`xfem`](https://mooseframework.inl.gov/docs/PRs/20254/coverage/xfem)

</details>



This comment will be updated on new commits.

joshuahansel committed:
Made hydraulic diameter materials take material property name

Refs #20253

joshuahansel committed:
Created hydraulic diameter material property in FlowChannel1Phase

Different flow models may have different definitions of hydraulic
diameter and also may or may not use automatic differentiation.

Refs #20253

joshuahansel committed:
Used AD hydraulic diameter in AD materials

Closes #20253
